src/Features/Core/Portable/ReplacePropertyWithMethods/AbstractReplacePropertyWithMethodsService.cs

```diff
@@ -24,11 +24,17 @@ internal abstract class AbstractReplacePropertyWithMethodsService<TIdentifierNam
 
         protected static SyntaxNode GetFieldReference(SyntaxGenerator generator, IFieldSymbol propertyBackingField)
         {
-            var through = propertyBackingField.IsStatic
-                ? generator.TypeExpression(propertyBackingField.ContainingType)
-                : generator.ThisExpression();
+            var memberName = generator.IdentifierName(propertyBackingField.Name);
+            if (propertyBackingField.IsStatic)
+            {
+                return propertyBackingField.ContainingType == null
```

Comment:
Can we at least add a comment that we don't understand why this is possible? Has the compiler team looked to make sure there's no obvious races here?